core/src/main/java/com/linecorp/armeria/common/RequestHeadersBuilder.java

```diff
@@ -39,20 +44,54 @@
     RequestHeadersBuilder method(HttpMethod method);
 
     /**
-     * Sets the {@code ":path"} headers.
+     * Sets the {@code ":path"} header.
      */
     RequestHeadersBuilder path(String path);
 
     /**
-     * Sets the {@code ":scheme"} headers.
+     * Sets the {@code ":scheme"} header.
      */
     RequestHeadersBuilder scheme(String scheme);
 
     /**
-     * Sets the {@code ":authority"} headers.
+     * Sets the {@code ":scheme"} header from the specified {@link SessionProtocol}.
+     */
+    default RequestHeadersBuilder scheme(SessionProtocol sessionProtocol) {
+        requireNonNull(sessionProtocol, "sessionProtocol");
+        switch (sessionProtocol) {
+            case HTTPS:
+            case H2:
+            case H1:
+                scheme("https");
+                break;
+            case HTTP:
+            case H2C:
+            case H1C:
+                scheme("http");
+                break;
+            default:
```

Comment:
for a rainy day: test of usage which triggers this (not unit test, like end-to-end).

Reply:
Added tests! :rainbow:

Reply:
Ah, it's impossible to trigger this in an integration test because ctx.sessioProtocol() never returns .